### PR TITLE
Expand documentation, testing, and functionality of ValueTrackers; remove ExponentialValueTracker

### PR DESCRIPTION
**Pull request by SwiddisZwei**
## Changelog / Overview

Added more documentation and inline operators to ValueTracker and ComplexValueTracker. Brought coverage for value_tracker.py to 100%. Removed ExponentialValueTracker.

## Motivation

Using ValueTracker is a bit cumbersome, and some testing and documentation is missing. This is especially true for Complex value trackers. This adds more inline operators beyond `__iadd__` and `__isub__`, and also expanded the documentation and testing for the trackers.

## Explanation for Changes

The following inline operations are now valid. Additionally, trackers can now be used in boolean expressions by checking the boolean cast of their value.
```py
tracker = ValueTracker(1.0)
tracker *= 2.0
tracker /= 2.0
tracker //= 2.0
tracker %= 4.0
tracker **= 4.0
```

## Testing Status

Passes all new tests locally, on windows.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by kolibril13** [comment]
Thanks for this pr!
Right now I am a bit busy, but I will test and review this pr on the weekend.

File: manim/mobject/value_tracker.py
Comment (behackl) [inline manim/mobject/value_tracker.py:194]: Same comment regarding the note on `.animate`. It might make sense to point out that complex values are actually stored as point coordinates (i.e., a + jb is stored internally as [a, b, 0]).

**Review by behackl — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've left a bunch of suggestions, mainly concerning the short summaries in the docstrings.

From the implementation of `ExponentialValueTracker` I've seen above, I don't really see a reason to keep it around and I think it should be deprecated and removed -- what do you think about that?

I like the added example for `ComplexValueTracker`, good job!

**Comment by kolibril13** [comment]
> From the implementation of `ExponentialValueTracker` I've seen above, I don't really see a reason to keep it around and I think it should be deprecated and removed -- what do you think about that?

+1 for deprecating the `ExponentialValueTracker`. Only partly related to this pr, but regarding ValueTrackers in general, I think it would be really nice to have a DeltaValutracker some time in future, see: https://github.com/ManimCommunity/manim/issues/1290

**Review by huguesdevimeux — COMMENTED** [review]
overall SGTM. 
+1 for deprecating

File: manim/mobject/value_tracker.py
Comment (SwiddisZwei) [reply manim/mobject/value_tracker.py:167]: I don't think it makes sense to provide those implementation details to the user. It's useful from a development standpoint, but a user shouldn't need to know that it's dependent on an internal array.

Comment (SwiddisZwei) [reply manim/mobject/value_tracker.py:170]: After thinking about it, I think it does make sense to distinctly mention the points array, because it can be useful as shorthand for using the complex value directly. Added documentation to the get_value method.

**Comment by kolibril13** [comment]
I will also remove the test for ExponentialValueTracker, hope that is ok for you, when I directly push on your branch, @SwiddisZwei

**Comment by kolibril13** [comment]
https://manimce--1382.org.readthedocs.build/en/1382/reference/manim.mobject.value_tracker.ValueTracker.html
image
The animate link seems to be broken, maybe you can find a way to link it here correctly:
https://manimce--1382.org.readthedocs.build/en/1382/reference/manim.mobject.mobject.Mobject.html#manim.mobject.mobject.Mobject.animate

File: manim/mobject/value_tracker.py
Comment (kolibril13) [inline manim/mobject/value_tracker.py:77]:
```suggestion
        """Get the current value of this ValueTracker."""
```

File: manim/mobject/value_tracker.py
Comment (kolibril13) [inline manim/mobject/value_tracker.py:95]:
```suggestion
        """Add ``d_value`` to the value of this ValueTracker."""
```

File: manim/mobject/value_tracker.py
Comment (kolibril13) [reply manim/mobject/value_tracker.py:77]: I just saw that it is all the time this value tracker, so we can also keep it like that.

File: tests/test_value_tracker.py
Comment (kolibril13) [inline tests/test_value_tracker.py:4]: I made a pull request to your branch to remove this here

File: manim/mobject/value_tracker.py
Comment (kolibril13) [inline manim/mobject/value_tracker.py:141]: Same comment as above

File: manim/mobject/value_tracker.py
Comment (behackl) [inline manim/mobject/value_tracker.py:80]: Please change all the `Union[int, float]` simply to `float`.

```suggestion
    def set_value(self, value: float):
```

**Review by behackl — CHANGES_REQUESTED** [review]
I've added one final comment. :-)

This PR can be merged after this and the current merge conflict is resolved. Thank you very much for your efforts!

File: manim/mobject/value_tracker.py
Comment (kolibril13) [inline manim/mobject/value_tracker.py:85]:
```suggestion
    def increment_value(self, d_value: float]):
```
and also for the ones beneath

**Review by kolibril13 — APPROVED** [review]

**Review by behackl — APPROVED** [review]

**Pull request merged**
